seven_dof_arm: 
  joint_state_controller:
    type: joint_state_controller/JointStateController
    publish_rate: 50  

  # Position Controllers ---------------------------------------
  joint1_position_controller:
    type: position_controllers/JointPositionController
    joint: shoulder_pan_joint
    pid: {p: 100.0, i: 0.01, d: 10.0}
  joint2_position_controller:
    type: position_controllers/JointPositionController
    joint: shoulder_pitch_joint
    pid: {p: 100.0, i: 0.01, d: 10.0}
  joint3_position_controller:
    type: position_controllers/JointPositionController
    joint: elbow_roll_joint
    pid: {p: 100.0, i: 0.01, d: 10.0}
  joint4_position_controller:
    type: position_controllers/JointPositionController
    joint: elbow_pitch_joint
    pid: {p: 100.0, i: 0.01, d: 10.0}
  joint5_position_controller:
    type: position_controllers/JointPositionController
    joint: wrist_roll_joint
    pid: {p: 100.0, i: 0.01, d: 10.0}
  joint6_position_controller:
    type: position_controllers/JointPositionController
    joint: wrist_pitch_joint
    pid: {p: 100.0, i: 0.01, d: 10.0}
  joint7_position_controller:
    type: position_controllers/JointPositionController
    joint: gripper_roll_joint
    pid: {p: 100.0, i: 0.01, d: 10.0}
  base_velocity_controller:
    type: "diff_drive_controller/DiffDriveController"
    left_wheel: ['left_wheel_hinge_back', 'left_wheel_hinge_front']
    right_wheel: ['right_wheel_hinge_back', 'right_wheel_hinge_front']
    publish_rate: 50
    pose_covariance_diagonal: [0.001, 0.001, 0.001, 0.001, 0.001, 0.03]
    twist_covariance_diagonal: [0.001, 0.001, 0.001, 0.001, 0.001, 0.03]
    cmd_vel_timeout: 0.5
    wheel_separation : 0.445208
    wheel_radius : 0.085
    base_frame_id: robot_footprint
    odom_frame_id: odom
    enable_odom_tf: true
    estimate_velocity_from_position: false
    wheel_separation_multiplier: 1.0 
    wheel_radius_multiplier    : 1.0 
    linear:
      x:
        has_velocity_limits    : true
        max_velocity           : 2.0   
        has_acceleration_limits: true
        max_acceleration       : 3.0   
    angular:
      z:
        has_velocity_limits    : true
        max_velocity           : 3.0   
        has_acceleration_limits: true
        max_acceleration       : 6.0   
    pid_gains:
        base_lfw: {p: 20.0, i: 0.01, d: 0.0, i_clamp: 0.0}
        base_lbw: {p: 20.0, i: 0.01, d: 0.0, i_clamp: 0.0}
        base_rfw: {p: 20.0, i: 0.01, d: 0.0, i_clamp: 0.0}
        base_lbw: {p: 20.0, i: 0.01, d: 0.0, i_clamp: 0.0}


